code/game/objects/items/weapons/RCD.dm

```diff
@@ -597,7 +619,18 @@
  * * amount - an amount of matter to check for
  */
 /obj/item/rcd/proc/checkResource(amount, mob/user)
-	return matter >= amount
+	. = matter >= amount
+	if(!. && user)
+		to_chat(user, no_ammo_message)
+		flick("[icon_state]_empty", src)
+
+	return .
+
+/obj/item/rcd/update_icon()
+	..()
+	var/ratio = CEILING((matter / max_matter) * 10, 1)
+	cut_overlays()
+	add_overlay("[icon_state]_charge[ratio]")
 
```

Comment:
We tend to split these into their various sub-procs these days (`update_overlays()` in this case), I think the same can be done here. Then you neeed to update all of your `update_icon()` calls with `update_icon(UPDATE_OVERLAYS)`

Reply:
yes this works, I'll do this.

Trying to wrap my head around this though. What is the actual meaningful difference between `update_overlays()` and `update_icon()` if they both do the exact same thing

Reply:
It's going to affect you less here, but it means we can have granular control over what we want to update, saving CPU. No need to rebuild an entire human icon when you just want to remove a blood overlay, for example.